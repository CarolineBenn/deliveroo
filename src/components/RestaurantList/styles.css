@media (min-width: 678px) {
  .restaurantList {
    margin: 3rem 0;
  }
}

.restaurantList:first-child {
  border-top: 1px solid #E5E5E5;
}