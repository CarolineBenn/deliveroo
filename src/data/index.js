export const restaurants = [
  {
    id: 1,
    name: 'Bobo Social',
    tags: ['Vietnamese'],
    priceCategory: '££',
    time: '30',
    permalink: '/bobo-social'
  },
  {
    id: 2,
    name: 'Tsunami',
    tags: ['Vietnamese'],
    priceCategory: '££',
    time: '30',
    permalink: 'tsunami'
  },
  {
    id: 3,
    name: 'Wild Game Co',
    tags: ['Vietnamese'],
    priceCategory: '££',
    time: '30',
    permalink: 'wild-game-co'
  }
];

export const restaurant = {
  name: 'Bobo Social',
  tags: ['Burgers'],
  priceCategory: '££',
  blurb: 'Serving the highest-quality burgers and sides with an array of creative and tasty toppings, Bobo Social cooks its rare-breed beed in a special oven to take you to burger heaven!',
  menu: [
    {
      category: 'Burgers',
      options: [
        {
          id: 1,
          name: 'Bobo',
          desc: '6oz rare-breed beef, lettuce, tomatoes, charcoal-roasted red onions, BOBO sauce',
          price: '£8.95',
        },
        {
          id: 2,
          name: 'Bobo with Cheese',
          desc: '6oz rare-breed beef, Westcombe Cheddar, lettuce, tomatoes, charcoal-roasted red onions, BOBO sauce',
          price: '£9.95',
        }
      ],  
    }
  ],
};


export const neighbourhood = {
  name: 'Hampstead',
  description: 'Hampstead is a great place to while away a whole day. As well as three of London’s biggest and best museums, there’re plenty of other things to do that are sure to entertain and amuse everyone in your group.',
  count: 34,
  restaurants: restaurants,
};
