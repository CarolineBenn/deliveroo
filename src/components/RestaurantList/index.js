import React from 'react';
import PropTypes from 'prop-types';

import RestaurantListItem from '../RestaurantListItem';

import './styles.css';

const RestaurantList = ({ restaurants }) => (
  <ul className="unstyledList restaurantList">
    {restaurants.map(restaurant => (
      <RestaurantListItem key={restaurant.id} {...restaurant} />
    ))}
  </ul>
);

export default RestaurantList;

RestaurantList.propTypes = {
  /**
   * Array of establishments serving neighbourhood.
   */
  restaurants: PropTypes.arrayOf(PropTypes.shape({
    /**
     * Numerical id of restaurant.
     */
    id: PropTypes.number.isRequired,
    /**
     * String name of restaurant.
     */
    name: PropTypes.string.isRequired,
    /**
     * Array of tags for restaurant (e.g. ['Vietnamese', 'Noodles']).
     */
    tags: PropTypes.arrayOf(PropTypes.string),
    /**
     * String indication of restuarant price.
     */
    priceCategory: PropTypes.string.isRequired,
    /**
     * Estimated delivery time.
     */
    time:  PropTypes.string.isRequired,
    /**
     * String link to restaurant page.
     */
    permalink:  PropTypes.string.isRequired,
  })).isRequired,
};
