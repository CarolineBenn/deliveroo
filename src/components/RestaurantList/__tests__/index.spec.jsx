import React from 'react';
import renderer from 'react-test-renderer';

import { neighbourhood } from '../../../data';

import RestaurantList from '..';

// Mock components that have their own snapshot tests
jest.mock('../../RestaurantListItem', () => 'RestaurantListItem');

test('RestaurantList renders correctly', () => {
  const tree = renderer.create(
    <RestaurantList restaurants={neighbourhood.restaurants} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});