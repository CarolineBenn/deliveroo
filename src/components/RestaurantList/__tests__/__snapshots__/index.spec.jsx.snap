// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList renders correctly 1`] = `
<ul
  className="unstyledList restaurantList"
>
  <RestaurantListItem
    id={1}
    name="Bobo Social"
    permalink="/bobo-social"
    priceCategory="££"
    tags={
      Array [
        "Vietnamese",
      ]
    }
    time="30"
  />
  <RestaurantListItem
    id={2}
    name="Tsunami"
    permalink="tsunami"
    priceCategory="££"
    tags={
      Array [
        "Vietnamese",
      ]
    }
    time="30"
  />
  <RestaurantListItem
    id={3}
    name="Wild Game Co"
    permalink="wild-game-co"
    priceCategory="££"
    tags={
      Array [
        "Vietnamese",
      ]
    }
    time="30"
  />
</ul>
`;
